mflowgen/tile_array/custom-dc-postcompile/custom-dc-postcompile.tcl

```diff
@@ -0,0 +1,10 @@
+#=========================================================================
+# Post-compile DC commands file
+#=========================================================================
+
+# We are manually ungrouping all references to mantle_wire* modules after 
+# compilation because doing so before compilation caused hi,lo -> tile_id
+# connection on pe and memory tiles to be optimized away and replaced with 
+# tie cells.
+set_dont_touch [get_references *mantle_wire*] false
+ungroup [get_references *mantle_wire*]
```

Comment:
What does ungrouping do?

Reply:
ungroup essentially eliminates these hierarchies from the design. In this case, it removes all the mantle_wire instances, leaving only the wires they contained.

Reply:
I was also curious so I looked it up in the manual, found this:

> Ungroups the specified hinsts. Ungrouping dissolves the hinsts and moves the contents into its
> parent directory. By default, an hinst or an inst inside one of the ungrouped hinsts is renamed by
> concatenating its name to its parent's name with “_”. for example, “ungroup h1”, would rename h1/i1
> to h1_i1. Name collisions are resolved by appending _1, _2, etc. until a unique name is found. If a
> given hinst cannot be ungrouped, a warning is issued about why it cannot be ungrouped.
>